### PR TITLE
[SPARK-21149][R] Add job description API for R

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

Extend `setJobDescription` to SparkR API.

## How was this patch tested?

It looks difficult to add a test. Manually tested as below:

```r
df <- createDataFrame(iris)
count(df)
setJobDescription("This is an example job.")
count(df)
```

prints ...

### CODE REVIEW

File: R/pkg/R/context.R [hunk 1]
```diff
@@ -295,6 +295,23 @@ setCheckpointDirSC <- function(sc, dirName) {
...
+#' Set a description that is shown as a job description in UI.
+#'
+#' @rdname setJobDescription
+#' @param value The job description of the current job.
+#' @export
+#' @examples
+#'\dontrun{
+#' setJobDescription("This is a example job.")
+#'}
+#' @note setJobDescription since 2.3.0
```
Comment (HyukjinKwon) [inline R/pkg/R/context.R]: Just for sure,

**Comment by HyukjinKwon** [comment]
cc @felixcheung, could you take a look and see if it is as you intended in the JIRA?

File: R/pkg/R/context.R [hunk 2]
```diff
@@ -295,6 +295,23 @@ setCheckpointDirSC <- function(sc, dirName) {
...
+#' Set a description that is shown as a job description in UI.
+#'
+#' @rdname setJobDescription
+#' @param value The job description of the current job.
+#' @export
+#' @examples
+#'\dontrun{
+#' setJobDescription("This is an example job.")
+#'}
+#' @note setJobDescription since 2.3.0
```
Comment (HyukjinKwon) [inline R/pkg/R/context.R]: Just for sure,

File: R/pkg/NAMESPACE [hunk 3]
```diff
@@ -403,6 +403,7 @@ export("as.DataFrame",
        "refreshTable",
        "setCheckpointDir",
        "setCurrentDatabase",
+       "setJobDescription",
```
Comment (felixcheung) [inline R/pkg/NAMESPACE]: I think this should go to https://github.com/HyukjinKwon/spark/blob/c591e63d5937403bb462bdf76795af8e3f324aca/R/pkg/NAMESPACE#L78

File: R/pkg/R/context.R [hunk 4]
```diff
@@ -295,6 +295,23 @@ setCheckpointDirSC <- function(sc, dirName) {
...
+setJobDescription <- function(value) {
```
Comment (felixcheung) [inline R/pkg/R/context.R]: and this should probably go to https://github.com/apache/spark/blob/422aa67d1bb84f913b06e6d94615adb6557e2870/R/pkg/R/sparkR.R#L536

**Comment by HyukjinKwon** [comment]
Thank you @felixcheung. I checked docs and the same thing in the PR description.

**Review by felixcheung — APPROVED** [review]
LGTM.
thinking more on this, let's add a call to it in test - that would at least detect signature changes etc.

https://github.com/apache/spark/blob/9f4ff9552470fb97ca38bb56bbf43be49a9a316c/R/pkg/tests/fulltests/test_context.R#L102

**Comment by HyukjinKwon** [comment]
@felixcheung, thanks for guiding me.

**Review by felixcheung — APPROVED** [review]
LGTM pending appveyor

**Comment by felixcheung** [comment]
merged to master, thanks!

**Pull request closed without merging**